source/common/grpc/google_async_client_impl.cc

```diff
@@ -42,7 +42,8 @@ void GoogleAsyncClientThreadLocal::completionThread() {
     const GoogleAsyncTag::Operation op = google_async_tag.op_;
     GoogleAsyncStreamImpl& stream = google_async_tag.stream_;
     ENVOY_LOG(trace, "completionThread CQ event {} {}", op, ok);
-    std::unique_lock<std::mutex> lock(stream.completed_ops_lock_);
+    Locking::MutexLock lock(stream.completed_ops_lock_);
```

Comment:
Have you verified that TSAN will catch concurrency bugs by manually adding a bug here and observing the compiler output? We had the issue originally when I first added these annotations that they went untested until we did the Google import, at which point we noticed real issues for the first time.

Reply:
I verified that omitting locks result in compile errors with this change.  In fact that was how I found the existing annotations to switch to absl::Mutex.  TSAN I think is orthogonal.